src/dbnode/storage/index/types.go

```diff
@@ -79,6 +88,18 @@ type QueryResults struct {
 	Exhaustive bool
 }
 
+// AggregationOptions enables users to specify constraints on aggregations.
+type AggregationOptions struct {
+	StartInclusive time.Time
+	EndExclusive   time.Time
+	Limit          int
+	FilterFn       TagNameFilterFn
+	Type           AggregationType
+}
+
+// TagNameFilterFn provides a predicate using which aggregation results can be filtered.
+type TagNameFilterFn func(tagName []byte) bool
```

Comment:
Just wondering how would we go about serializing this function to send to the index? Unless you're thinking we'd do the filtering in the client side rather than the index side?